### ISSUE TITLE
Source version in Model

### ISSUE THREAD

**Issue by bastiencarreres** [sncosmo/sncosmo#352]
Hi, I'm working on SN Ia simulation and I would keep track of the Source version that I used in some meta-file to re-load the same model when needed. I have remarked that the `class Model(_ModelBase)` has not a `version` arg that can be passed to `get_source`. Is it something that can be implemented in a future version?

**Comment by cwwalter** [comment]
Also tagging @jchiang87.

**Comment by cwwalter** [comment]
Tagging @JoanneBogart.

**Comment by benjaminrose** [comment]
@bastiencarreres, yeah, I think we should add this. This allows people to use `sncosmo.Model` with a string, but still fully define their source file. Currently, you would need to create a full `sncosmo.Source` object to do this. There are not many sources with multiple versions, so this has not come up before, but DESC is also asking for this. I think we should add a dictionary and not mess with the order of the input variables. I think we should do something like this:

```
def __init__(self, source, effects=None,
                effect_names=None, effect_frames=None, source_params={}):
       # Set source and add source parameter names
       self._source = get_source(source, copy=True, **source_params)
```

Then you can use `snc.models._SOURCES.retrieve('salt2', source_params={'version': '2.4'})`. It is a bit more verbose, but helps keep the default process smoother & backwards compatible.


@JoanneBogart, with this patch, we can add a similar version value to skyCatalogs's class [`SNObject`](https://github.com/LSSTDESC/skyCatalogs/blob/4ad5371c109a3a4a8b51200c49876defe97d56e9/python/desc/skycatalogs/utils/sn_tools.py#L17).

**Comment by MickaelRigault** [comment]
Hello guys, I'm also interested with this functionality. The proposition from @bastiencarreres looks like the good one to me. 
We want to be able to do 
```python
model = sncosmo.Model("salt2", version="2.4")
```
or a parsable string (that may helps the dev as no new variables)
```python
model = sncosmo.Model("salt2 v=2.4")
```
or (and this would be cleaner from the start: forbid the fact that sources may have the same Name (so rename "salt2"->salt2.4, salt2T21, salt2B22 etc)
```python
model = sncosmo.Model("salt2.4")
```
